docs/developer/extending/apps/building-payment-app.mdx

```diff
@@ -0,0 +1,856 @@
+
+Then, declare the subscription query for the `TRANSACTION_PROCESS_SESSION` webhook:
+
+```ts
+import { TransactionProcessSessionPayloadFragment } from "../../../../generated/graphql"; // Import the generated payload type
+import { gql } from "urql";
+
+// 💡 Remember to regenerate the types after adding the new subscription query
+const TransactionProcessSessionPayload = gql`
+  fragment TransactionProcessSessionPayload on TransactionProcessSession {
+    action {
+      amount
+      currency
+      actionType
+    }
+  }
+`;
+
+const TransactionProcessSessionSubscription = gql`
+  ${TransactionProcessSessionPayload}
+  subscription TransactionProcessSession {
+    event {
+      ...TransactionProcessSessionPayload
+    }
+  }
+`;
+```
+
+And then, create the `SaleorSyncWebhook` instance, as well as the webhook handler:
+
+```ts
+import { TransactionProcessSessionPayloadFragment } from "../../../../generated/graphql"; // Import the generated payload type
+import { gql } from "urql";
+// highlight-start
+import { SaleorSyncWebhook } from "@saleor/app-sdk/handlers/next";
+import { saleorApp } from "../../../saleor-app";
+// highlight-end
+
+// 💡 Remember to regenerate the types after adding the new subscription query
+const TransactionProcessSessionPayload = gql`
+  fragment TransactionProcessSessionPayload on TransactionProcessSession {
+    action {
+      amount
+      currency
+      actionType
+    }
+  }
+`;
+
+const TransactionProcessSessionSubscription = gql`
+  ${TransactionProcessSessionPayload}
+  subscription TransactionProcessSession {
+    event {
+      ...TransactionProcessSessionPayload
+    }
+  }
+`;
+
+// highlight-start
+export const transactionProcessSessionWebhook =
+  new SaleorSyncWebhook<TransactionProcessSessionPayloadFragment>({
+    name: "Transaction Process Session",
+    webhookPath: "/api/webhooks/transaction-process-session",
+    event: "TRANSACTION_PROCESS_SESSION",
+    apl: saleorApp.apl,
+    query: TransactionProcessSessionSubscription,
+  });
+
+export default transactionProcessSessionWebhook.createHandler(
+  (req, res, ctx) => {
+    const { payload, event, baseUrl, authData } = ctx;
+
+    console.log("Transaction Process Session payload:", payload);
+
+    return res.status(200).json({
+      result: "CHARGE_SUCCESS",
+      amount: payload.action.amount,
+      pspReference: crypto.randomUUID(),
+    });
+  }
+);
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+// highlight-end
+```
+
+Once again, our handler is simple. We only log the payload and return the `CHARGE_SUCCESS` event status, no questions asked.
+
+:::info
+
+A real-world `TRANSACTION_PROCESS_SESSION` webhook handler might:
+
+- Call the payment provider API to confirm the payment status.
+- Return either `CHARGE_SUCCESS` or `CHARGE_FAILURE` based on the response.
```

Comment:
or `CHARGE_ACTION_REQUIRED`. Payment provider could have a multiple validation checks to be perform before returning SUCCESS or FAILURE. So for some payment provider, ``TRANSACTION_PROCESS_SESSION` could be called in the "loop"

Reply:
✅